compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -241,14 +255,24 @@ public void visit(BLangResource resourceNode) {
 
         validateReturnParameter(resourceNode, null, false);
         validateReferences(resourceNode);
+        validateDeprecationDocumentation(resourceNode.markdownDocumentationAttachment, false, resourceNode.pos, false);
+        validateDeprecatedParametersDocumentation(resourceNode.markdownDocumentationAttachment, resourceNode.pos);
     }
 
     private void validateDeprecationDocumentation(BLangMarkdownDocumentation documentation,
-                                                  boolean isDeprecationAnnotationAvailable, DiagnosticPos pos) {
+                                                  boolean isDeprecationAnnotationAvailable,
+                                                  DiagnosticPos pos, boolean isExpected) {
         if (documentation == null) {
             return;
         }
 
+        BLangMarkDownDeprecationDocumentation deprecationDocumentation =
+                documentation.getDeprecationDocumentation();
+        if (!isExpected && deprecationDocumentation != null) {
+            dlog.error(pos, DiagnosticCode.DEPRECATION_DOCUMENTATION_NOT_ALLOWED);
```

Comment:
There is a validation below to check if the deprecation annotation is available right? And the annotation specifies on what kind of constructs it can be used.

So, IMO we don't need this check. Since it'll fail below anyway.

Reply:
Yes, that validation is there and it says `invalid documentation: ''Deprecated'' documentation is only allowed on constructs annotated as ''@deprecated''`

```
# This is a test service
#
# # Deprecated parameters
# + x - deprecated x
# # Deprecated
service s on new Listener() {

} 
```
But in this example, without the above validation, it says `invalid documentation: ''Deprecated'' documentation is only allowed on constructs annotated as ''@deprecated''`, with the above validation it says `''Deprecated'' documentation is not allowed here`.

Reply:
The issue IMO is that we have two point to validate.

We are trying to do manually part of the validation that is based on the annotation.